<!DOCTYPE html>
<html lang="en-us" dir="ltr">
<head>
  <meta property="og:title" content="" />
<meta property="og:description" content="Calling Variants on Sequencing Data #  Before we dive into one of the nf-core pipelines used for variant calling, it&rsquo;s worth looking at some theoretical aspects of variant calling.
Overview #  The term &ldquo;variant calling&rdquo; is rooted in the history of DNA sequencing, and it indicates an approach where we identify (i.e. call) positions in a genome (loci) which are variable in a population (genetic variants). The specific genotype of an individual at that variant locus is then assigned." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://example.org/tutorial/1.theory/" /><meta property="article:section" content="tutorial" />

<meta property="og:site_name" content="ðŸ““ Amethyst" />

<title>1.Theory | ðŸ““ Amethyst</title>
<link rel="manifest" href="/manifest.json">
<link rel="icon" href="/favicon.png" type="image/x-icon">
<link rel="stylesheet" href="/book.min.5a2664b4517d55ec16199b62d0903d0484859e7f9aded9023c0b399e6a03e519.css" integrity="sha256-WiZktFF9VewWGZti0JA9BISFnn&#43;a3tkCPAs5nmoD5Rk=" crossorigin="anonymous">

<!--
Made with Book Theme
https://github.com/alex-shpak/hugo-book
-->
  
  



<meta charset="UTF-8" />
<meta
  name="description"
  content="Calling Variants on Sequencing Data #  Before we dive into one of the nf-core pipelines used for variant calling, it&rsquo;s worth looking at some theoretical aspects of variant calling.
Overview #  The term &ldquo;variant calling&rdquo; is rooted in the history of DNA sequencing, and it indicates an approach where we identify (i.e. call) positions in a genome (loci) which are variable in a population (genetic variants). The specific genotype of an individual at that variant locus is then assigned."
/>
<title>
  ðŸ““ Amethyst
</title>
<meta name="viewport" content="width=device-width, initial-scale=1" />






<link rel="shortcut icon" type="image/png"  href="http://example.org//favicon.png" />

























  
  


<link href="http://example.org/styles.e08ceb33360cec132feb69cfb982e2a4.min.css" rel="stylesheet" />


<link href="http://example.org/quartz/styles/_light_syntax.86a48a52faebeaaf42158b72922b1c90.min.css" rel="stylesheet" id="theme-link">

  

<link rel="preload" href="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.css" as="style"
      onload="this.onload=null;this.rel='stylesheet'"
      integrity="sha384-R4558gYOUz8mP9YWpZJjofhk+zx0AS11p36HnD2ZKj/6JR5z27gSSULCNHIRReVs" crossorigin="anonymous">
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.js"
        integrity="sha384-z1fJDqw8ZApjGO3/unPWUPsIymfsJmyrDVWC8Tv/a1HeOtGmkwNd/7xUS0Xcnvsx"
        crossorigin="anonymous"></script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/contrib/auto-render.min.js"
        integrity="sha384-+XBljXPPiv+OzfbB3cVmLHf4hdUFHlWNZN5spNQ7rmHTXpd7WvJum6fIACpNNfIR"
        crossorigin="anonymous"></script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.2/dist/contrib/copy-tex.min.js"
        integrity="sha384-ww/583aHhxWkz5DEVn6OKtNiIaLi2iBRNZXfJRiY1Ai7tnJ9UXpEsyvOITVpTl4A"
        crossorigin="anonymous"></script>

  






<script src="http://example.org/quartz/js/darkmode.2dde094b8728e01ab3f1666d1d15dd1b.min.js"></script>



<script src="http://example.org/quartz/js/util.c69e233dc1cc331a30b0f670e657b425.min.js"></script>



<script async src="https://unpkg.com/@floating-ui/core@0.7.3"></script>
<script async src="https://unpkg.com/@floating-ui/dom@0.5.4"></script>

<script src="http://example.org/quartz/js/popover.f03552ccb84d99ca615d1cfb9abde59e.min.js"></script>




<script defer src="http://example.org/quartz/js/code-title.ce4a43f09239a9efb48fee342e8ef2df.min.js"></script>




<script defer src="http://example.org/quartz/js/clipboard.2913da76d3cb21c5deaa4bae7da38c9f.min.js"></script>




<script defer src="http://example.org/quartz/js/callouts.7723cac461d613d118ee8bb8216b9838.min.js"></script>



 
<script>
  const SEARCH_ENABLED =  false 
  const LATEX_ENABLED =  true 
  const PRODUCTION =  true 
  const BASE_URL = "http://example.org/"
  const fetchData = Promise.all([
        fetch("http:\/\/example.org\/indices\/linkIndex.f6425e3c9c5a78cb1a57d6271bd0c88d.min.json")
          .then(data => data.json())
          .then(data => ({
            index: data.index,
            links: data.links,
          })),
        fetch("http:\/\/example.org\/indices\/contentIndex.56753ef950db2332e50166167e5b613e.min.json")
          .then(data => data.json()),
      ])
      .then(([{index, links}, content]) => ({
        index,
        links,
        content,
      }))

    const render = () => {
    
    const siteBaseURL = new URL(BASE_URL);
    const pathBase = siteBaseURL.pathname;
    const pathWindow = window.location.pathname;
    const isHome = pathBase == pathWindow;

    addCopyButtons();
    

    

    addCollapsibleCallouts();
    

    
    initPopover(
      "http://example.org",
       false 
    )
    

    

    
      var els = document.getElementsByClassName("mermaid");
      if (els.length > 0) {
        import('https://unpkg.com/mermaid@9/dist/mermaid.esm.min.mjs').then(
          (obj) => {
            
            
            obj.default.init();
          }
        )
      }
    
  }

  const init = (doc = document) => {
    
    addCopyButtons();
    

    addTitleToCodeBlocks();
    renderMathInElement(doc.body, {
      delimiters: [
        {left: '$$', right: '$$', display: true},
        {left: '$', right: '$', display: false},
      ],
      macros: {
        'â€™': "'"
      },
      throwOnError : false
    });
    
  };
</script>

<script>
  window.Million = {
    navigate: (url) => (window.location.href = url),
    prefetch: () => {},
  }

  window.addEventListener("DOMContentLoaded", () => {
    init()
    render()
  })
</script>


</head>
<div id="search-container">
  <div id="search-space">
    <input autocomplete="off" id="search-bar" name="search" type="text" aria-label="Search"
      placeholder="Search for something..." dir="">
    <div id="results-container">
    </div>
  </div>
</div>

<script src="https://cdn.jsdelivr.net/npm/flexsearch@0.7.21/dist/flexsearch.bundle.js"
  integrity="sha256-i3A0NZGkhsKjVMzFxv3ksk0DZh3aXqu0l49Bbh0MdjE=" crossorigin="anonymous" defer></script>

<script defer src="http://example.org/quartz/js/full-text-search.e6e2e0c213187ca0c703d6e2c7a77fcd.min.js"></script>



<header class="book-header">
    
    <label for="menu-control">
      <img src="/svg/menu.svg" class="book-icon" alt="Menu" />
    </label>
    <h1 id="page-title"><a href="http://example.org/">ðŸ““ Amethyst</a></h1>
    <div class="spacer"></div>
    <div id="search-icon" class="quartz-search">
      <p>Search</p>
      <svg tabindex="0" aria-labelledby="title desc" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 19.9 19.7"><title id="title">Search Icon</title><desc id="desc">Icon to open search</desc><g class="search-path" fill="none"><path stroke-linecap="square" d="M18.5 18.3l-5.4-5.4"/><circle cx="8" cy="8" r="7"/></g></svg>
    </div>
    <div class='darkmode'>
    <input class='toggle' id='darkmode-toggle' type='checkbox' tabindex="-1">
    <label id="toggle-label-light" for='darkmode-toggle' tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="dayIcon" x="0px" y="0px" viewBox="0 0 35 35" style="enable-background:new 0 0 35 35;" xml:space="preserve">
            <title>Light Mode</title>
            <path d="M6,17.5C6,16.672,5.328,16,4.5,16h-3C0.672,16,0,16.672,0,17.5    S0.672,19,1.5,19h3C5.328,19,6,18.328,6,17.5z M7.5,26c-0.414,0-0.789,0.168-1.061,0.439l-2,2C4.168,28.711,4,29.086,4,29.5    C4,30.328,4.671,31,5.5,31c0.414,0,0.789-0.168,1.06-0.44l2-2C8.832,28.289,9,27.914,9,27.5C9,26.672,8.329,26,7.5,26z M17.5,6    C18.329,6,19,5.328,19,4.5v-3C19,0.672,18.329,0,17.5,0S16,0.672,16,1.5v3C16,5.328,16.671,6,17.5,6z M27.5,9    c0.414,0,0.789-0.168,1.06-0.439l2-2C30.832,6.289,31,5.914,31,5.5C31,4.672,30.329,4,29.5,4c-0.414,0-0.789,0.168-1.061,0.44    l-2,2C26.168,6.711,26,7.086,26,7.5C26,8.328,26.671,9,27.5,9z M6.439,8.561C6.711,8.832,7.086,9,7.5,9C8.328,9,9,8.328,9,7.5    c0-0.414-0.168-0.789-0.439-1.061l-2-2C6.289,4.168,5.914,4,5.5,4C4.672,4,4,4.672,4,5.5c0,0.414,0.168,0.789,0.439,1.06    L6.439,8.561z M33.5,16h-3c-0.828,0-1.5,0.672-1.5,1.5s0.672,1.5,1.5,1.5h3c0.828,0,1.5-0.672,1.5-1.5S34.328,16,33.5,16z     M28.561,26.439C28.289,26.168,27.914,26,27.5,26c-0.828,0-1.5,0.672-1.5,1.5c0,0.414,0.168,0.789,0.439,1.06l2,2    C28.711,30.832,29.086,31,29.5,31c0.828,0,1.5-0.672,1.5-1.5c0-0.414-0.168-0.789-0.439-1.061L28.561,26.439z M17.5,29    c-0.829,0-1.5,0.672-1.5,1.5v3c0,0.828,0.671,1.5,1.5,1.5s1.5-0.672,1.5-1.5v-3C19,29.672,18.329,29,17.5,29z M17.5,7    C11.71,7,7,11.71,7,17.5S11.71,28,17.5,28S28,23.29,28,17.5S23.29,7,17.5,7z M17.5,25c-4.136,0-7.5-3.364-7.5-7.5    c0-4.136,3.364-7.5,7.5-7.5c4.136,0,7.5,3.364,7.5,7.5C25,21.636,21.636,25,17.5,25z" />
        </svg>
    </label>
    <label id="toggle-label-dark" for='darkmode-toggle' tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="nightIcon" x="0px" y="0px" viewBox="0 0 100 100" style="enable-background='new 0 0 100 100'" xml:space="preserve">
            <title>Dark Mode</title>
            <path d="M96.76,66.458c-0.853-0.852-2.15-1.064-3.23-0.534c-6.063,2.991-12.858,4.571-19.655,4.571  C62.022,70.495,50.88,65.88,42.5,57.5C29.043,44.043,25.658,23.536,34.076,6.47c0.532-1.08,0.318-2.379-0.534-3.23  c-0.851-0.852-2.15-1.064-3.23-0.534c-4.918,2.427-9.375,5.619-13.246,9.491c-9.447,9.447-14.65,22.008-14.65,35.369  c0,13.36,5.203,25.921,14.65,35.368s22.008,14.65,35.368,14.65c13.361,0,25.921-5.203,35.369-14.65  c3.872-3.871,7.064-8.328,9.491-13.246C97.826,68.608,97.611,67.309,96.76,66.458z" />
        </svg>
    </label>
</div>
</header>


<body dir="ltr">
  <input type="checkbox" class="hidden toggle" id="menu-control" />
  <main class="flex">
    <aside class="book-menu">
      <div class="book-menu-content">
        
  <nav>
<div class="menu-search">
  <div id="search-icon" class="quartz-search">
    <p>Search</p>
    <svg tabindex="0" aria-labelledby="title desc" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 19.9 19.7">
      <title id="title">Search Icon</title>
      <desc id="desc">Icon to open search</desc>
      <g class="search-path" fill="none">
        <path stroke-linecap="square" d="M18.5 18.3l-5.4-5.4" />
        <circle cx="8" cy="8" r="7" />
      </g>
    </svg>
  </div>
</div>













  



  
  <ul>
    
      
        <li>
          
  
  

  
    <span class="book-menu-title">Getting Started</span>
  

          
  <ul>
    
      
        <li>
          
  
  

  
    <a href="/tutorial/1.theory/" class="active">1.Theory</a>
  

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/tutorial/2.sarek/" class="">2.Sarek</a>
  

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/tutorial/3.interpretation/" class="">3.Interpretation</a>
  

        </li>
      
    
  </ul>

        </li>
      
    
  </ul>











  
<ul>
  
  <li>
    <a href="https://github.com/lescai-teaching/sarek-tutorial"  target="_blank" rel="noopener">
        Github
      </a>
  </li>
  
</ul>






</nav>




  <script>(function(){var a=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(b){localStorage.setItem("menu.scrollTop",a.scrollTop)}),a.scrollTop=localStorage.getItem("menu.scrollTop")})()</script>


 
      </div>
    </aside>
    <div class="book-page container">
      

      
      
  <article class="markdown">
    <h1 class="title">1.Theory</h1>
    






  
  

  
  

  
  

  
  

  
  

  
  

  
  















  
  
  
  
  
  
  
  
  
  






<h1 id="calling-variants-on-sequencing-data">
  Calling Variants on Sequencing Data
  <a class="anchor" href="#calling-variants-on-sequencing-data">#</a>
</h1>
<p>Before we dive into one of the nf-core pipelines used for variant calling, it&rsquo;s worth looking at some theoretical aspects of variant calling.</p>
<h2 id="overview">
  Overview
  <a class="anchor" href="#overview">#</a>
</h2>
<p>The term &ldquo;variant calling&rdquo; is rooted in the history of DNA sequencing, and it indicates an approach where we identify (i.e. call) positions in a genome (loci) which are variable in a population (genetic variants). The specific genotype of an individual at that variant locus is then assigned.</p>
<p>There are many different approaches for calling variants from sequencing data: here, we will look more specifically at a reference-based variant calling approach, i.e. where a reference genome is needed and variant sites are identified by comparing the reads to this reference.</p>
<p>Over the years, also thanks to the work carried out by the 
<a href="https://gatk.broadinstitute.org/hc/en-us" rel="noopener">GATK team</a> at the Broad Institute, there has been a convergence on a &ldquo;best practices&rdquo; workflow, which is summarised in the diagram below:</p>
<p>


<img src="http://example.org/tutorial/./img/overview.excalidraw.svg" width="auto" alt="overview"  /></p>
<p>In this scheme we can identify a few key phases in the workflow. Pre-processing is the first part, where raw data are handled and mapped to a genome reference, to be then transformed in order to increase the accuracy of the following analyses. Then, variant calling is carried out. This is followed by filtering and annotation.
Here we will briefly discuss these key steps, which might vary depending on the specific type of data one is performing variant calling on.</p>
<h2 id="alignment">
  Alignment
  <a class="anchor" href="#alignment">#</a>
</h2>
<p>The alignment step is where reads obtained from genome fragments of a sample are identified as originating from a specific location in the genome.<br>
This step is essential in a reference-based workflow, because it is the comparison of the raw data with the reference to inform us on whether a position in the genome might be variable or not.</p>
<p>Mismatches, insertions and deletions (INDELs) as well as duplicated regions make this step sometimes challenging: this is the reason why an appropriate aligner has to be chosen, depending on the sequencing application and data type.</p>
<p>Once each raw read has been aligned to the region of the genome it is most likely originating from, the sequence of all reads overlapping each locus can be used to identify potentially variable sites. Each read will support the presence of an allele identical to the reference, or a different one (alternative allele), and the variant calling algorithm will measure the weighted support for each allele.</p>
<p>However, the support given by the raw data to alternative variants might be biased. For this reason, one can apply certain corrections to the data to ensure the support for the alleles is assessed correctly. This is done by performing the two steps described below: marking duplicates, and recalibrating base quality scores.</p>
<h2 id="marking-duplicates">
  Marking Duplicates
  <a class="anchor" href="#marking-duplicates">#</a>
</h2>
<p>Duplicates are non-independent measurements of a sequence fragment.</p>
<p>Since DNA fragmentation is theoretically random, reads originating from different fragments provide independent information. An algorithm can use this information to assess the support for different alleles.
When these measurements however are not independent, the raw data might provide a biased support towards a specific allele.</p>
<p>Duplicates can be caused by PCR during library preparation (library duplicates) or might occur during sequencing, when the instrument is reading the signal from different clusters (as in Illumina short read sequencing). These latter are called &ldquo;optical duplicates&rdquo;.</p>
<p>A specific step called &ldquo;marking duplicates&rdquo; identifies these identical pairs using their orientation and 5' position (before any clipping), which will be assumed to be coming from the same input DNA template: one representative pair is then chosen based on quality scores and other criteria, while the other ones are marked.
Marked reads are then ignored in the following steps.</p>
<h2 id="base-quality-score-recalibration">
  Base Quality Score Recalibration
  <a class="anchor" href="#base-quality-score-recalibration">#</a>
</h2>
<p>Among the parameters used by a variant calling algorithm to weigh the support for different alleles, the quality score of the base in the read at the variant locus is quite important.
Sequencing instruments, however, can make systematic errors when reading the signal at each cycle, and cannot account for errors originated in PCR.</p>
<p>Once a read has been aligned to the reference, an appropriate algorithm can however compare the error rate estimated from the existing base quality scores, with the actual differences observed with the reference sequence (empirical quality), and perform appropriate corrections.
This process is called &ldquo;base quality score recalibration&rdquo; (BQSR).</p>
<p>To calculate empirical qualities, the algorithm simply counts the number of mismatches in the observed bases. Any mismatch which does not overlap a known variant is considered an error. The empirical error rate is simply the ratio between counted errors and the total observed bases.
A Yates correction is applied to this, to avoid either dividing by 0 or dealing with small counts.</p>
<p>$$
e_{empirical} = \frac{n_{mismatches} + 1}{n_{bases} +2}
$$</p>
<p>The empirical error is expressed as a Quality in Phred-scale:</p>
<p>$$
Q_{empirical} = -10 \times log_{10}(e_{empirical})
$$</p>
<p>Let&rsquo;s use a simple example like the one in the diagram below, where for illustrative purposes we only consider the bases belonging to the same read.</p>
<p>


<img src="http://example.org/tutorial/./img/bqsr.excalidraw.svg" width="auto" alt="bqsr"  /></p>
<p>In this example we have 3 mismatches, but one is a reported variant site: we therefore only count 2 errors, over 10 observed bases. According to the approach we just explained,</p>
<p>$$
Q_{empirical} = -10 \times log_{10}(\frac{2 + 1}{10 +2}) = 6.29
$$</p>
<p>To calculate the average reported Q score, we should sum the error probabilities and then convert them back into phred scale:</p>
<p>$$
Q_{average} = -10 \times log_{10}(\frac {0.1 + 0.1 + 0.01 + 0.1 + 0.01 + 0.01 + 0.01 + 0.1 + 0.1 + 0.1}{10}) = 11.94
$$</p>
<p>Our empirical Q score would be 6.29, the average reported Q score is 11.94, and therefore the $\Delta = 11.94 - 6.29 = 5.65$</p>
<p>The recalibrated Q score of each base would correspond to the reported Q score minus this $\Delta$.</p>
<p>In a real sequencing dataset, this calculation is performed for different groups (bins) of bases: those in the same lane, those with the same original quality score, per machine cycle, per sequencing context.
In each bin, the difference ($\Delta$) between the average reported quality and the empirical quality is calculated.
The recalibrated score would then be the reported score minus the sum of all deltas calculated in each bin the base belongs to.</p>
<p>A detailed summary of this approach can be found on the 
<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890531-Base-Quality-Score-Recalibration-BQSR-" rel="noopener">GATK BQSR page</a>. We also found quite useful this 
<a href="https://rstudio-pubs-static.s3.amazonaws.com/64456_4778547202f24f32b0edc325e96b061a.html" rel="noopener">step by step guide</a> through the matematical approach. Full details are explained in the 
<a href="https://www.nature.com/articles/ng.806" rel="noopener">publication</a> that first proposed this method.</p>
<h2 id="calling-variants">
  Calling Variants
  <a class="anchor" href="#calling-variants">#</a>
</h2>
<p>Once we have prepared the data for an accurate identification of the variants, we are ready to perform the next steps.
The most important innovation introduced some years ago in this part of the workflow, has been to separate the identification of a variant site (i.e. variant calling itself) from the assignment of the genotype to each individual.
This approach makes the computation more approachable, especially for large sample cohorts: BAM files are only accessed per-sample in the first step, while multi-sample cohort data are used together in the second step in order to increase the accuracy of genotype assignment.</p>
<h3 id="identifying-variants">
  Identifying Variants
  <a class="anchor" href="#identifying-variants">#</a>
</h3>
<p>In this phase, which is performed on each sample independently, a first step uses a sliding window to count differences compared to the reference (i.e. mismatches, INDELs) and potentially variable regions are identified. GATK calls these &ldquo;active regions&rdquo;.
Then, a local graph assembly of the reads is created to identify plausible haplotypes, which are aligned to the reference with a traditional alignment algorithm called &ldquo;Smith-Waterman&rdquo;: this is used to identify variants.
For each read in an active region, the support for each of the haplotypes is counted and a likelihood score for each combination of read/haplotype is calculated.
The likelihoods at this step allow to calculate the support for each of the alleles in a variant site, and read-haplotype likelihoods are a key input for the Bayesian statistics used to determine the most likely genotype.
This first genotype assignment could be sufficient if one analysed a single sample only.</p>
<h3 id="assigning-genotypes">
  Assigning Genotypes
  <a class="anchor" href="#assigning-genotypes">#</a>
</h3>
<p>When multiple samples are analysed, information from each of them could collectively improve the genotype assignment.
This is because the magnitude of potential biases (example: strand bias) can be better estimated, and because the distributions of those annotations used to inform the genotype assignment become more stable when more data are available, by combining multiple samples.
The use of a larger cohort also increases the sensitivity.</p>
<p>This is possible if the variant calling step is run by producing a variation of the VCF file format called GVCF: this format includes, in addition to variant sites, also non-variant intervals in the genome of each sample. Moreover, it reports probability likelihoods of a non-reference symbolic allele at these non-variant intervals.
This information allows to re-genotype each sample by using data from the whole cohort.</p>
<p>You can read more on the GATK website about the 
<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890431-The-logic-of-joint-calling-for-germline-short-variants" rel="noopener">logic of joint calling</a>.</p>
<h3 id="filtering-variants">
  Filtering Variants
  <a class="anchor" href="#filtering-variants">#</a>
</h3>
<p>There are several ways to spot potential false positives through filtering.</p>
<p><em>Hard filtering</em> uses pre-defined thresholds of different variant annotations (allele-depth, mapping quality and many others) in order to flag variants passing all these criteria, and those failing to meet any of them. This approach is mostly useful when calling a few samples and enough data are not available for more sophisticated solutions.</p>
<p><em>Soft filtering</em> infers the thresholds to be applied from the data themselves. This approach uses the distributions of the annotations, and their overlap with known and validated variants: it defines those combinations of annotations which are more likely to describe true positives (the variants they refer to in the analysis cohort overlap with those validated in other databases). This approach is used by a GATK tool called Variant Quality Score Recalibration (VQSR).</p>
<p>More details can be found on the 
<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531612-Variant-Quality-Score-Recalibration-VQSR-" rel="noopener">GATK VQSR page</a>.</p>
<p>More recently, pre-trained deep learning models are also available to filter variants based on neural network architectures trained on a large number of variants from population databases.</p>
<h2 id="annotation">
  Annotation
  <a class="anchor" href="#annotation">#</a>
</h2>
<p>Once the analysis has produced a final VCF file, the final step which is necessary to interpret the results is called &ldquo;annotation&rdquo;.
This step uses different databases to describe (annotate) each variant from a genomic, biological, or population point of view.
The software used to carry out this task will add information to the VCF file such as:</p>
<ul>
<li>the gene each variant overlaps with</li>
<li>the transcript the variant overlaps with</li>
<li>the potential biological consequence on each of those transcripts</li>
<li>population frequency (minor allele frequency, described in different databases such as gnomAD)</li>
</ul>
<p>And several other items we can use to interpret our findings from a biological or clinical point of view.</p>


    

<hr/>


<div class="page-end" id="footer">
    <div class="backlinks-container">
        <h3>Backlinks</h3>
<ul class="backlinks">
    
    
    
    
    
    
    
    <li>
      No backlinks found
    </li>
    
</ul>

    </div>
    <div>
        <script
  src="https://cdn.jsdelivr.net/npm/d3@6.7.0/dist/d3.min.js"
  integrity="sha256-+7jaYCp29O1JusNWHaYtgUn6EhuP0VaFuswhNV06MyI="
  crossorigin="anonymous"
></script>
<h3>Interactive Graph</h3>
<div id="graph-container"></div>
<style>
  :root {
    --g-node: var(--secondary);
    --g-node-active: var(--primary);
    --g-node-inactive: var(--visited);
    --g-link: var(--outlinegray);
    --g-link-active: #5a7282;
  }
</style>

<script src="http://example.org/quartz/js/graph.cbd78cfa87df7d3e230d16fc24f06548.js"></script>

    </div>
</div>






<div id="contact_buttons">
    <footer>
        
        
        <p>Made with <a href="https://github.com/64bitpandas/amethyst">Amethyst</a>, Â© 2024 Ben Cuan</p>
        <ul>
            
            <li><a href="http://example.org/">Home</a></li>
            <li><a href="https://github.com/lescai-teaching">Teaching Repos</a></li><li><a href="http://elearning.unipv.it">e-learning platform</a></li></ul>
    </footer>
</div>


    
  </article>
 
      
      
      
      
  
 

      <label for="menu-control" class="hidden book-menu-overlay"></label>
    </div>

    
    <aside class="book-toc">
      <div class="book-toc-content">
        
  
<nav id="TableOfContents">
  <ul>
    <li><a href="#overview">Overview</a></li>
    <li><a href="#alignment">Alignment</a></li>
    <li><a href="#marking-duplicates">Marking Duplicates</a></li>
    <li><a href="#base-quality-score-recalibration">Base Quality Score Recalibration</a></li>
    <li><a href="#calling-variants">Calling Variants</a>
      <ul>
        <li><a href="#identifying-variants">Identifying Variants</a></li>
        <li><a href="#assigning-genotypes">Assigning Genotypes</a></li>
        <li><a href="#filtering-variants">Filtering Variants</a></li>
      </ul>
    </li>
    <li><a href="#annotation">Annotation</a></li>
  </ul>
</nav>


 
      </div>
    </aside>
    
  </main>

  
</body>
</html>












